File: content/search/source/includes/index/field-types/vector/code-snippets/python/simple_index.py
from pymongo import MongoClient
from pymongo.operations import SearchIndexModel

# connect to your Atlas deployment
uri = "<connection-string>"
client = MongoClient(uri)

# set namespace
database = client["sample_mflix"]
collection = database["embedded_movies"]

# define your MongoDB Search index
search_index_model = SearchIndexModel(
    definition={
        "mappings": {
            "dynamic": True,
            "fields": {
                "plot_embedding_voyage_3_large": {
                    "numDimensions": 2048,
                    "quantization": "scalar",
                    "similarity": "dotProduct",
                    "type": "vector"
                }
            }
        },
    },
    name="default",
)

# create the index
result = collection.create_search_index(model=search_index_model)
print(f"New index name: {result}")

File: content/search/source/includes/index/field-types/vector/code-snippets/nodejs/simple_index.js
const { MongoClient } = require("mongodb");

// connect to your Atlas deployment
const uri =  
  "<connection-string>";

const client = new MongoClient(uri);

async function run() {
  try {

    // set namespace
    const database = client.db("sample_mflix");
    const collection = database.collection("embedded_movies");
    
    // define your MongoDB Search index
    const index = {
        name: "default",
        definition: {
            /* search index definition fields */
            "mappings": {
                "dynamic": true,
                "fields": {
                    "plot_embedding_voyage_3_large": {
                        "numDimensions": 2048,
                        "quantization": "scalar",
                        "similarity": "dotProduct",
                        "type": "vector"
                    }
                }
            }
        }
    }

    // run the helper method
    const result = await collection.createSearchIndex(index);
    console.log("New index name: " + result);
  } finally {
    await client.close();
  }
}

run().catch(console.dir);